dev/src/bulk-writer.ts

```diff
@@ -648,14 +789,20 @@ export class BulkWriter {
    *   console.log('Executed all writes');
    * });
    */
-  close(): Promise<void> {
+  async close(): Promise<void> {
```

Comment:
Public API methods shouldn't use the `async` keyword since it messes with the execution order. `this.closed = true;` should be set synchronously. If you make this method `async`, it executes one Promise tick later.